src/main/java/com/kobylynskyi/graphql/codegen/java/JavaGraphQLTypeMapper.java

```diff
@@ -122,4 +124,22 @@ public ValueMapper getValueMapper() {
         return valueMapper;
     }
 
+    @Override
+    public List<String> getJacksonTypeIdAnnotations(MappingContext mappingContext, boolean isUnion) {
+        List<String> defaults = new ArrayList<>();
+        if (Boolean.TRUE.equals(mappingContext.getGenerateJacksonTypeIdResolver()) && isUnion) {
+            defaults.add("com.fasterxml.jackson.annotation.JsonTypeInfo(use = " +
+                    "com.fasterxml.jackson.annotation.JsonTypeInfo.Id.NAME, property = \"__typename\")");
+            String modelPackageName = DataModelMapper.getModelPackageName(mappingContext);
+            if (modelPackageName == null) {
+                modelPackageName = "";
+            } else if (Utils.isNotBlank(modelPackageName)) {
+                modelPackageName += ".";
+            }
+            defaults.add("com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver(" + modelPackageName +
+                    "GraphqlJacksonTypeIdResolver.class)");
+        }
+        return defaults;
```

Comment:
@gdhordain
There are around 10 lines of duplicate code in 3 classes (for each language)
So I propose to create an abstract method only for returning a JsonTypeIdResolver (so far I see there's a difference inly in this line:
```java
@Override
public List<String> getJsonTypeIdResolverAnnotationForPackage(String modePackageName) {
    return "com.fasterxml.jackson.databind.annotation.JsonTypeIdResolver("
        + modelPackageName + "GraphqlJacksonTypeIdResolver.class)";
}
```
Please let me know what you think. Thanks.

Reply:
Yes, I didn't think too much about it at the weekend.

Reply:
I pushed a refactor to avoid these duplicated lines.